### PR TITLE
fix: to #157015 fix view label command localized

### LINKED ISSUE #157015: Focus on __ View commands are improperly localized
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.70-insider
- OS Version: Windows 11

Steps to Reproduce:

1. Configure display language
2. Search for 'Focus on' in the command palette (all views get a focus command auto-registered)
3. :bug: builtin views have the category and view title translated in the original version of the title being displayed
image

We should probably support `ILocalizedString` for view titles and categories when registering these commands

### PR DESCRIPTION
**Pull request by yiliang114**
Fixes #157015

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
Awesome!

Looks like you do have a couple of issues still:
```
[tsec-compile-check      ] src/vs/workbench/contrib/scm/browser/scm.contribution.ts(115,2): error TS2322: Type 'string' is not assignable to type 'ILocalizedString'.
[tsec-compile-check      ] src/vs/workbench/contrib/userDataSync/browser/userDataSyncViews.ts(181,32): error TS2352: Conversion of type '{ id: string; name: string; ctorDescriptor: SyncDescriptor<TreeViewPane>; when: RawContextKey<boolean>; canToggleVisibility: true; canMoveView: false; treeView: TreeView; collapsed: false; hideByDefault: false; }' to type 'ITreeViewDescriptor' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
[tsec-compile-check      ]   Types of property 'name' are incompatible.
[tsec-compile-check      ]     Type 'string' is not comparable to type 'ILocalizedString'.
[tsec-compile-check      ] 
[tsec-compile-check      ] 
[tsec-compile-check      ] Found 2 errors.
[tsec-compile-check      ] error Command failed with exit code 1.
```
From the CI

File: src/vs/workbench/contrib/quickaccess/browser/viewQuickAccess.ts [hunk 1]
```diff
@@ -104,7 +104,7 @@ export class ViewQuickAccessProvider extends PickerQuickAccessProvider<IViewQuic
 			for (const view of viewContainerModel.allViewDescriptors) {
 				if (this.contextKeyService.contextMatchesRules(view.when)) {
 					result.push({
-						label: view.name,
+						label: view.name.original,
```
Comment (yiliang114) [inline src/vs/workbench/contrib/quickaccess/browser/viewQuickAccess.ts:107]: Use original here to display original text when other places are localized

**Comment by yiliang114** [comment]
> Awesome!
> 
> Looks like you do have a couple of issues still:
> 
> ```
> [tsec-compile-check      ] src/vs/workbench/contrib/scm/browser/scm.contribution.ts(115,2): error TS2322: Type 'string' is not assignable to type 'ILocalizedString'.
> [tsec-compile-check      ] src/vs/workbench/contrib/userDataSync/browser/userDataSyncViews.ts(181,32): error TS2352: Conversion of type '{ id: string; name: string; ctorDescriptor: SyncDescriptor<TreeViewPane>; when: RawContextKey<boolean>; canToggleVisibility: true; canMoveView: false; treeView: TreeView; collapsed: false; hideByDefault: false; }' to type 'ITreeViewDescriptor' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
> [tsec-compile-check      ]   Types of property 'name' are incompatible.
> [tsec-compile-check      ]     Type 'string' is not comparable to type 'ILocalizedString'.
> [tsec-compile-check      ] 
> [tsec-compile-check      ] 
> [tsec-compile-check      ] Found 2 errors.
> [tsec-compile-check      ] error Command failed with exit code 1.
> ```
> 
> From the CI

Thanks!

Now it is reasonable to say that all the processing has been completed.
image

(I think I should have handled all the compilation errors before. When I update the main branch code, some code may be restored due to automatic merging)

File: src/vs/workbench/contrib/editSessions/browser/editSessionsViews.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export class EditSessionsDataViews extends Disposable {
 		const viewsRegistry = Registry.as<IViewsRegistry>(Extensions.ViewsRegistry);
 		viewsRegistry.registerViews([<ITreeViewDescriptor>{
 			id: viewId,
-			name,
+			name: { value: name, original: name },
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/editSessions/browser/editSessionsViews.ts:50]: This one should be updated by either making `EDIT_SESSIONS_TITLE` an `ILocalizedString` or have this be:

```ts
name: { value: name, original: '......' },
```

File: src/vs/workbench/contrib/quickaccess/browser/viewQuickAccess.ts (hunk 1 above)
Comment (TylerLeonhardt) [reply src/vs/workbench/contrib/quickaccess/browser/viewQuickAccess.ts:107]: Not sure I quite understand what you mean here. Why original? (also could you add a comment saying why)

File: src/vs/workbench/contrib/userDataSync/browser/userDataSyncViews.ts [hunk 3]
```diff
@@ -180,7 +180,7 @@ export class UserDataSyncDataViews extends Disposable {
 		const viewsRegistry = Registry.as<IViewsRegistry>(Extensions.ViewsRegistry);
 		viewsRegistry.registerViews([<ITreeViewDescriptor>{
 			id,
-			name,
+			name: { value: name, original: name },
```
Comment (TylerLeonhardt) [inline src/vs/workbench/contrib/userDataSync/browser/userDataSyncViews.ts:183]: this is using name for both.

File: src/vs/workbench/services/userDataProfile/browser/userDataProfileImportExportService.ts [hunk 4]
```diff
@@ -992,7 +992,7 @@ export class UserDataProfileImportExportService extends Disposable implements IU
 		const actionRunner = new ActionRunner();
 		const descriptor: ITreeViewDescriptor = {
 			id,
-			name,
+			name: { value: name, original: name },
```
Comment (TylerLeonhardt) [inline src/vs/workbench/services/userDataProfile/browser/userDataProfileImportExportService.ts:995]: this is using name for both.

**Review by TylerLeonhardt — COMMENTED** [review]
This is super close! Just a few comments

File: src/vs/workbench/services/userDataProfile/browser/userDataProfileImportExportService.ts (hunk 4 above)
Comment (yiliang114) [reply src/vs/workbench/services/userDataProfile/browser/userDataProfileImportExportService.ts:995]: The name here is passed in through the `showProfilePreviewView` function and looks like a user-defined profile name. Does this value determine that name is different from original ?

File: src/vs/workbench/browser/parts/views/viewsService.ts [hunk 5]
```diff
@@ -485,7 +485,7 @@ export class ViewsService extends Disposable implements IViewsService {
 	private registerFocusViewAction(viewDescriptor: IViewDescriptor, category?: string | ILocalizedString): IDisposable {
 		return registerAction2(class FocusViewAction extends Action2 {
 			constructor() {
-				const title = localize({ key: 'focus view', comment: ['{0} indicates the name of the view to be focused.'] }, "Focus on {0} View", viewDescriptor.name);
+				const title = localize({ key: 'focus view', comment: ['{0} indicates the name of the view to be focused.'] }, "Focus on {0} View", viewDescriptor.name.value);
 				super({
 					id: viewDescriptor.focusCommand ? viewDescriptor.focusCommand.id : `${viewDescriptor.id}.focus`,
 					title: { original: `Focus on ${viewDescriptor.name} View`, value: title },
```
Comment (TylerLeonhardt) [inline src/vs/workbench/browser/parts/views/viewsService.ts:491]: I think you need to change this to `viewDescriptor.name.original` no? This is actually the fix you're fixing. You tested this fix, right?

Comment (yiliang114) [reply src/vs/workbench/browser/parts/views/viewsService.ts]: > I think you need to change this to `viewDescriptor.name.original` right? Otherwise you will see `[Object object]` You may want to spot check the other references to make sure you are using the correct value.

Yes, yes, I also found this problem and have already disposed of it. Now it behaves like this in the Chinese environment:

image

**Comment by TylerLeonhardt** [comment]
FYI we are currently in the middle of a release right now, so I want to hold off on getting this in til next week. That way we can let it sit in Insiders for a bit to make sure we don't cause any regressions due to the number of files changed. I think the PR is ready, but I'll approve and merge early next week.

File: src/vs/workbench/services/views/test/browser/viewDescriptorService.test.ts [hunk 6]
```diff
@@ -60,19 +60,19 @@ suite('ViewDescriptorService', () => {
 			{
```
[the remark this answers is not in the document]
Comment (yiliang114) [reply src/vs/workbench/services/views/test/browser/viewDescriptorService.test.ts:60]: Thanks for the help!

**Comment by TylerLeonhardt** [comment]
@yiliang114 I hope you don't hate me for this but I just added a new thing that will make this PR much smaller and have less duplicate strings.

I have added a `localize2` function [in this PR](https://github.com/microsoft/vscode/pull/194750) that allows us to replace all of these:
```ts
{ value: nls.localize({ comment: ['Debug is a noun in this context, not a verb.'], key: 'debugPanel' }, "Debug Console"), original: 'Debug Console' }
```
with a much simpler:
```ts
nls.localize2({ comment: ['Debug is a noun in this context, not a verb.'], key: 'debugPanel' }, "Debug Console")
```

this function returns an `ILocalizedString` and handles the original string correctly.

Can you move over to using that so the code is easier to maintain and we don't need to have two of the same string next to each other?

**Comment by yiliang114** [comment]
> @yiliang114 I hope you don't hate me for this but I just added a new thing that will make this PR much smaller and have less duplicate strings.
> 
> I have added a `localize2` function [in this PR](https://github.com/microsoft/vscode/pull/194750) that allows us to replace all of these:
> 
> ```ts
> { value: nls.localize({ comment: ['Debug is a noun in this context, not a verb.'], key: 'debugPanel' }, "Debug Console"), original: 'Debug Console' }
> ```
> 
> with a much simpler:
> 
> ```ts
> nls.localize2({ comment: ['Debug is a noun in this context, not a verb.'], key: 'debugPanel' }, "Debug Console")
> ```
> 
> this function returns an `ILocalizedString` and handles the original string correctly.
> 
> Can you move over to using that so the code is easier to maintain and we don't need to have two of the same string next to each other?
> 
> I think this won't apply to _everything_ in your PR but probably about 80-90% if the cases.

Haha, it looks better this way. Let me try again.

**Comment by yiliang114** [comment]
Please help me review the code again when you are free @TylerLeonhardt , I did not encountered an error  when I compiled locally.

image

**Review by TylerLeonhardt — APPROVED** [review]
LGTM thanks for taking this on!

File: src/vs/workbench/services/views/test/browser/viewContainerModel.test.ts [hunk 7]
```diff
@@ -63,13 +64,13 @@ suite('ViewContainerModel', () => {
 	});
 
 	test('empty model', function () {
-		container = ViewContainerRegistry.registerViewContainer({ id: 'test', title: { value: 'test', original: 'test' }, ctorDescriptor: new SyncDescriptor(<any>{}) }, ViewContainerLocation.Sidebar);
+		container = ViewContainerRegistry.registerViewContainer({ id: 'test', title: nls.localize2('test', 'test'), ctorDescriptor: new SyncDescriptor(<any>{}) }, ViewContainerLocation.Sidebar);
```
Comment (roblourens) [inline src/vs/workbench/services/views/test/browser/viewContainerModel.test.ts:67]: These will be excluded from the actual localization process, right?

**Review by roblourens — APPROVED** [review]

File: src/vs/workbench/services/views/test/browser/viewContainerModel.test.ts (hunk 7 above)
Comment (TylerLeonhardt) [reply src/vs/workbench/services/views/test/browser/viewContainerModel.test.ts:67]: Yep!
